src/citrine/resources/gemd_resource.py

```diff
@@ -0,0 +1,92 @@
+"""Collection class for generic GEMD objects and templates."""
+from typing import Type, Union, Optional, List, Iterator
+from uuid import UUID
+
+from gemd.entity.base_entity import BaseEntity
+from gemd.entity.link_by_uid import LinkByUID
+
+from citrine.resources.data_concepts import DataConcepts, DataConceptsCollection
+from citrine._session import Session
+
+
+class GEMDResourceCollection(DataConceptsCollection[DataConcepts]):
```

Comment:
The most consistent class name is probably `GEMDEntity` since that's the parent element of both object and template in gemd-python, though that includes several classes that are not directly supported with this class.

Reply:
I agree that "Entity" is the most appropriate term, if we're inheriting terminology from `gemd-python`. Doesn't it include everything? It looks like the things in `gemd-python` that inherit `BaseEntity` are just objects, object templates, and attribute templates.

The term "Entity" is used in some places in `citrine-python`, mostly in types as part of docstrings, so I wouldn't be opposed to calling this `GEMDEntityCollection`. On the other hand, `citrine-python` heavily uses the term "Resource" to refer to objects that are stored on-platform. The directory is `/resources`, and we also talk about "AI Resources" in places.

Reply:
I'm falling back to `resource` being the most fitting term within `citrine-python`. It is used significantly throughout the other collections, and seems general enough. I will opt to keep as such for this iteration.